lib/cloak/migrator.ex

```diff
@@ -9,6 +9,13 @@ defmodule Cloak.Migrator do
   def migrate(repo, schema) when is_atom(repo) and is_atom(schema) do
     validate(repo, schema)
 
+    case repo.aggregate(schema, :count, :id) do
+      0 -> {:ok, "Empty schema. No data migrated."}
+      _ -> migrate_schema_with_data(repo, schema)
+    end
```

Comment:
I can totally do that. My driver for making this modification was so that I could add a test to the regression suite to ensure that this functionality remains, as the lack of coverage I would guess made it easy for this to be added but then removed in a refactor.

Let me know your thoughts. I'll remove if the value of not changing the signature is more valuable than the test coverage. Alternatively, if you have any suggestions for another way I could add test coverage for this, I'm all ears.

Reply:
Sorry for the multiple comments, but I think to maintain the return value, I would keep the case statement, but modified to look like:

```elixir
case repo.aggregate(schema, :count, :id) do
      0 -> {:ok}
      _ -> migrate_schema_with_data(repo, schema)
    end
```

If I used the `unless`, wouldn't that change the return value to `nil`? Although, if we're fine with that, I could use that as the indicator in the test, though it's not obvious (to me) that'd be the only reason why it returns `nil`.

Happy to do whatever you think would be best. Thanks for the review.

Reply:
You're right about `unless`. Your solution is fine, except let's just return `:ok` and not `{:ok}`.

```elixir
case repo.aggregate(schema, :count, :id) do
  0 -> :ok
  _ -> migrate_schema_with_data(repo, schema)
end
```